frontend/src/components/common/header/header-container.tsx

```diff
@@ -0,0 +1,110 @@
+import { FC } from 'common/types/types';
+import { Link } from 'react-router-dom';
+import { FormEvent, MouseEvent, RefObject } from 'react';
+
+import { AppRoute, IconName } from 'common/enums/enums';
+import { MenuOptions } from 'common/enums/components';
+
+import styles from './header.module.scss';
+import { Icon } from '../icon';
+
+interface MenuOption {
+  type: MenuOptions;
+  text?: string;
+  icon: string;
+  onClick?: (e: MouseEvent) => void;
+}
+
+interface HeaderProps {
+  isLogged: boolean;
+  isMenuOpen: boolean;
+  searchValue: string;
+  handleClickUserMenu: (e: MouseEvent<HTMLButtonElement>) => void;
+  handleClickLogin(e: MouseEvent<HTMLElement>): void;
+  handleInputSearch(e: FormEvent<HTMLInputElement>): void;
+  options: MenuOption[];
+  userAvatar: string;
+  menuRef: RefObject<HTMLDivElement>;
+}
+
+const HeaderContainer: FC<HeaderProps> = ({
```

Comment:
according to pattern I sent you in 1st review - container is component with logic. Rename please. This component is render-only.